apps/src/turtle/turtle.js

```diff
@@ -697,13 +697,12 @@ Artist.prototype.evalCode = function(code) {
  * Set up this.code, this.interpreter, etc. to run code for editCode levels
  */
 Artist.prototype.generateTurtleCodeFromJS_ = function () {
+  var userCode = this.studioApp_.editor.getValue();
   this.code = dropletUtils.generateCodeAliases(dropletConfig, 'Turtle');
   this.userCodeStartOffset = this.code.length;
-  this.code += this.studioApp_.editor.getValue();
+  this.code += userCode;
   this.userCodeLength = this.code.length - this.userCodeStartOffset;
-
-  var session = this.studioApp_.editor.aceEditor.getSession();
-  this.cumulativeLength = codegen.aceCalculateCumulativeLength(session);
+  this.cumulativeLength = codegen.calculateCumulativeLength(userCode);
```

Comment:
these changes are not needed if you approve https://github.com/code-dot-org/code-dot-org/pull/6849 first